htdocs/societe/card.php

```diff
@@ -519,15 +519,15 @@
                             $error=$object->error; $errors=$object->errors;
                         }
                     }
-
-					// Customer categories association
-					$custcats = GETPOST( 'custcats', 'array' );
-					$object->setCategories($custcats, 'customer');
-
-					// Supplier categories association
-					$suppcats = GETPOST('suppcats', 'array');
-					$object->setCategories($suppcats, 'supplier');
-
+					if(!empty($user->rights->categorie->creer)){
```

Comment:
The permission categorie->creer is permission to add/delete a category among other categories). Right to admin the tags.
Here, we edit a category as a property of a thirdparty, so permission must be $user->rights->societe->creer and seems already tested before.

Reply:
Ok, my bad.
Here's my problem then : If a user hasn't right to read categorie, and update a thirdparty, it'll empty thirdparty's categories. So do you prefer to check $user->rights->categorie->lire ?

Reply:
I see.
Yes, just disable code if test on "$user->rights->categorie->lire "
Please add a comment to explain this case too so we understand why we have a test on read to update.